File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/Calendars/calendarproperties.rb
module Asposetasksjava
  module CalendarProperties
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        calendars = project.getCalendars().toList()

        i = 0
        while i < calendars.size()
          cal = calendars.get(i)
          if cal.getName() != nil
            puts "Base Calendar : "
            if cal.isBaseCalendar()
              puts "Self"
            else
              cal.getBaseCalendar().getName()
            end  
            puts "UID : " + cal.getUid().to_s
            puts "Name : " + cal.getName().to_s
          end
          i +=1
        end
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithCalendars/CalendarProperties.php
<?php
namespace Aspose\Tasks\WorkingWithCalendars;

use com\aspose\tasks\Project as Project;

class CalendarProperties{

    public static function run($dataDir=null)
    {

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $calendars = $project->getCalendars()->toList();

        $i = 0;
        while ($i < sizeof($calendars)) {
            $cal = $calendars->get($i);
            if ($cal->getName() != null) {
                print "Base Calendar : ";
                if ($cal -> isBaseCalendar()) {
                    print "Self";
                } else {
                    $cal->getBaseCalendar()->getName();
                }
                print "UID : " . (string)$cal->getUid();//.to_s
                print "Name : " . (string)$cal->getName();//.to_s
            }
            $i += 1;
        }
    }
}

?>